src/models/db/transfers.js

```diff
@@ -7,8 +9,32 @@ function * getTransfer (id, options) {
   return (yield Transfer.findById(id, options))
 }
 
+function * _upsertTransfer (transfer, options) {
+  assert(options.transaction)
+  // SQLite's implementation of upsert does not tell you whether it created the
+  // row or whether it already existed. Since we need to know to return the
+  // correct HTTP status code we unfortunately have to do this in two steps.
+  const existingTransfer = yield Transfer.findById(transfer.id, options)
+  if (existingTransfer) {
+    existingTransfer.setData(transfer)
+    yield existingTransfer.save(options)
+  } else {
+    yield Transfer.create(transfer, options)
+  }
+  return Boolean(existingTransfer)
+}
+
 function * upsertTransfer (transfer, options) {
-  yield Transfer.upsert(transfer, options)
+  if (options && options.transaction) {
+    return (yield _upsertTransfer(transfer, options))
```

Comment:
In some cases, it is `return yield <expression>` and in others `return (yield <expression>)`.